src/hidimstat/utils.py

```diff
@@ -9,38 +9,116 @@
 from torchmetrics import Accuracy
 
 
-def quantile_aggregation(pvals, gamma=0.5, gamma_min=0.05, adaptive=False):
+########################## quantile aggregation method ##########################
+def quantile_aggregation(pvals, gamma=0.05, n_grid=20, adaptive=False):
     """
-    This function implements the quantile aggregation method for p-values.
+    Implements the quantile aggregation method for p-values based on :cite:meinshausen2009p.
+
+    The function aggregates multiple p-values into a single p-value while controlling
+    the family-wise error rate. It supports both fixed and adaptive quantile aggregation.
+
+    Parameters
+    ----------
+    pvals : ndarray of shape (n_sampling*2, n_test)
+        Matrix of p-values to aggregate. Each row represents a sampling instance
+        and each column a hypothesis test.
+        p-values should be adjusted as defined by the equation 2.2 of :cite:meinshausen2009p.
+    gamma : float, default=0.05
+        Quantile level for aggregation. Must be in range (0,1].
+    n_grid : int, default=20
+        Number of grid points to use for adaptive aggregation. Only used if adaptive=True.
+    adaptive : bool, default=False
+        If True, uses adaptive quantile aggregation which optimizes over multiple gamma values.
+        If False, uses fixed quantile aggregation with the provided gamma value.
+
+    Returns
+    -------
+    ndarray of shape (n_test,)
+        Vector of aggregated p-values, one for each hypothesis test.
+
+    References
+    ----------
+    .. footbibliography::
+
+    Notes
+    -----
+    The aggregated p-values are guaranteed to be valid p-values in [0,1].
+    When adaptive=True, gamma is treated as the minimum gamma value to consider.
     """
     # if pvalues are one-dimensional, do nothing
     if pvals.shape[0] == 1:
         return pvals[0]
     if adaptive:
-        return _adaptive_quantile_aggregation(pvals, gamma_min)
+        return _adaptive_quantile_aggregation(pvals, gamma, n_grid=n_grid)
     else:
         return _fixed_quantile_aggregation(pvals, gamma)
 
 
-def fdr_threshold(pvals, fdr=0.1, method="bhq", reshaping_function=None):
-    if method == "bhq":
-        return _bhq_threshold(pvals, fdr=fdr)
-    elif method == "bhy":
-        return _bhy_threshold(pvals, fdr=fdr, reshaping_function=reshaping_function)
-    elif method == "ebh":
-        return _ebh_threshold(pvals, fdr=fdr)
-    else:
-        raise ValueError("{} is not support FDR control method".format(method))
+def _fixed_quantile_aggregation(pvals, gamma=0.5):
+    """
+    Quantile aggregation function based on :cite:meinshausen2009p
 
+    Parameters
+    ----------
+    pvals : 2D ndarray (n_sampling*2, n_test)
+        p-value (adjusted)
```

Comment:
From what I understand, adjusted refers to the equation (2.1) in “P -Values for High-Dimensional Regression.”. 
In the two-step procedure introduced by the authors:
 1. split the data in 2 a screening is run on the first set (aiming to reduce the dimension from d>n to d'<n). 
 2. Then, p-values are estimated using the second set. These p-values are then **adjusted** by a factor proportional to the cardinal of the ensemble obtained during the screening stage. This correction is important to control the FWER. 

 - Unless we implement precisely their method, I suggest that we remove the "(adjusted)" which is misleading. 
 - Should we implement their method?

_Originally posted by @jpaillard in https://github.com/mind-inria/hidimstat/pull/133#discussion_r1923241673_

Reply:
But we most often (if not always) use quantile aggregation  in cases where there is no data splitting, hence no 'adjusted' p-values. So we should rather drop the term because it's not clearly defined and actually meaningless in most situations.

Reply:
@jpaillard

Reply:
agreed

Reply:
I remove it.